pkg/utils/nadconf.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"encoding/json"
+
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	networkv1 "github.com/harvester/harvester-network-controller/pkg/apis/network.harvesterhci.io/v1beta1"
+)
+
+const (
+	KeyNetworkConf = networkv1.GroupName + "/layer3-network"
+)
+
+type Connectivity string
+
+const (
+	Connectable  Connectivity = "true"
+	Unconnetable Connectivity = "false"
+	Unknown      Connectivity = "unknown"
+)
+
+type Mode string
+
+const (
+	Auto   Mode = "auto"
+	Manual Mode = "manual"
+)
+
+type NadSelectedNetworks []nadv1.NetworkSelectionElement
+
+type Layer3NetworkConf struct {
+	Mode         Mode         `json:"mode"`
+	CIDR         string       `json:"cidr"`
+	Gateway      string       `json:"gateway"`
+	Connectivity Connectivity `json:"connectivity"`
```

Comment:
when the mode is auto, can the user specify the DHCP IP, also should we allow users to config the DNS as an optional option here?

Reply:
I wonder what is the DNS for here. DHCP IP? Do you mean the IP of the DHCP server?

Reply:
Yes the IP of the DHCP server, the DNS is redundant so please ignore here.

Reply:
I will try to sent DHCPINFORM message instead of DHCPDISCOVERY to get the DHCP server options.